### PR TITLE
TSQL: Add missing `HISTORY_RETENTION_PERIOD` sequence to the table option segment

### LINKED ISSUE #5271: [TSQL] Unparsable section when create table with `SYSTEM_VERSIONING`, parameter `HISTORY_RETENTION_PERIOD`
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Having an SQL script that create a table with the `SYSTEM_VERSIONING` and parameters `HISTORY_RETENTION_PERIOD` throws an unparsable section error.

### Expected Behaviour

We should not have an error of parsing when using this parameter.

### Observed Behaviour

According to the Microsoft documentation for the [CREATE TABLE](https://bit.ly/3RQazXQ), only `HISTORY_TABLE` and `DATA_CONSISTENCY_CHECK` are allowed as parameters. But also according to the documentation and examples provided in how to [manage historical data in temporal tables](https://bit.ly/3LMOyFA), we can provide the parameter `HISTORY_RETENTION_PERIOD`.

To fix this we just need to add it as optional in this file [dialect_tsql.py](https://github.com/sqlfluff/sqlfluff/blob/5680394b82888efc6d63656a48e049caf27029f8/src/sqlfluff/dialects/dialect_tsql.py#L1388C32-L1388C32).

### How to reproduce

```sql
CREATE TABLE [dbo].[PARSING_ERROR] (
	[id] INT NOT NULL
) ON [PRIMARY] WITH (
	SYSTEM_VERSIONING = ON (
		HISTORY_TABLE = [cpt].[CATEGORIES_HISTORY],
		HISTORY_RETENTION_PERIOD = INFINITE,
		DATA_CONSISTENCY_CHECK = ON
	)
);
```

### Dialect

tsql

### Version

```zsh
$ sqlfluff --version
sqlfluff, version 2.3.2

$ python --version
Python 3.11.5
```

### Configuration

Default configuration when running the following command:

```zsh
$ sqlfluff lint --dialect tsql file.sql
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by r-petit**
Adding fixtures for dialects.

### Brief summary of the change made

fixes #5271 

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by greg-finley** [comment]
Hey, thanks for the contribution! I see you put `wip` in the title. Do we need to wait for something before merging this?

File: src/sqlfluff/dialects/dialect_tsql.py [hunk 1]
```diff
@@ -1392,6 +1392,14 @@ class TableOptionSegment(BaseSegment):
                                     OneOf("ON", "OFF"),
                                     optional=True,
                                 ),
+                                Sequence(
```
Comment (greg-finley) [inline src/sqlfluff/dialects/dialect_tsql.py:1395]: Could `HISTORY_TABLE`, `DATA_CONSISTENCY_CHECK`, and `HISTORY_RETENTION_PERIOD` be given in any order? If so, I would put them in a `Delimited` which handles this and the `CommaSegment`

**Comment by r-petit** [comment]
> Hey, thanks for the contribution! I see you put `wip` in the title. Do we need to wait for something before merging this?

Yes, I'm working on it. Just saw that parameters aren't exactly the same as `Ref("DatetimeUnitSegment")`, I need to put custom list using `OneOf()`, but can't generate fixture yaml with new parameters, and don't know why.

File: src/sqlfluff/dialects/dialect_tsql.py (hunk 1 above)
Comment (r-petit) [reply src/sqlfluff/dialects/dialect_tsql.py:1395]: Yes it can, I'll correct this.

**Review by greg-finley — APPROVED** [review]
Love it, thanks!

**Comment by greg-finley** [comment]
Mind fixing the lint errors? Then we are good to go 🚀

**Pull request merged**